core/iwasm/compilation/aot.c

```diff
@@ -466,6 +511,19 @@ aot_create_comp_data(WASMModule *module, const char *target_arch,
     }
 #endif
 
+    if (!target_arch) {
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)            \
+    || defined(BUILD_TARGET_AARCH64) || defined(BUILD_TARGET_RISCV64_LP64D) \
+    || defined(BUILD_TARGET_RISCV64_LP64)
+        is_64bit_target = true;
+#endif
+    }
+    else {
+        if (strstr(target_arch, "64") != NULL) {
```

Comment:
Could we add comment about why we check it with "64"?

Reply:
Done